From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Sun, 14 Jul 2024 23:03:42 +0100
Subject: [PATCH 45/67] Drop AudioPlayback messages if no embedder element is
 defined

The AudioPlayback Window Actor passes "audio-playback" messages on to
the embedder element by calling it directly. If the embedder element is
null this will result in a "browser is null" error being output. This
doesn't do any real harm, but is unnecessary and could cause confusion.
This change adds a null check so that an attempt to call methods off the
browser only happen if it's defined.

On Sailfish OS the embedder element is always null, so there's no need
(or way) to pass these calls on anyway.
---
 toolkit/actors/AudioPlaybackParent.sys.mjs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/toolkit/actors/AudioPlaybackParent.sys.mjs b/toolkit/actors/AudioPlaybackParent.sys.mjs
index db682fd90b2b..e5b1b0245e4e 100644
--- a/toolkit/actors/AudioPlaybackParent.sys.mjs
+++ b/toolkit/actors/AudioPlaybackParent.sys.mjs
@@ -11,6 +11,9 @@ export class AudioPlaybackParent extends JSWindowActorParent {
   }
   receiveMessage(aMessage) {
     const browser = this.browsingContext.top.embedderElement;
+    if (!browser) {
+      return;
+    }
     switch (aMessage.name) {
       case "AudioPlayback:Start":
         this._hasAudioPlayback = true;
